### PR TITLE
Use asgs for masters workers

### PR DESCRIPTION
**Pull request by jeffb4**
Optionally use ASGs for master/worker nodes (defaults to ASG)

Additionally:
* fixes a bug with rancher version interpolation
* updates to latest stable rancher release for default (2.2.9)
* makes ssh key injection optional

### CODE REVIEW

File: rancher-server/data.tf [hunk 1]
```diff
@@ -47,11 +47,21 @@ data "helm_repository" "jetstack" {
...
+    values = [aws_autoscaling_group.rancher_master.0.name]
+  }
+}
+
+data "aws_instances" "rancher_worker" {
+  filter {
+    name   = "tag:aws:autoscaling:groupName"
+    values = [aws_autoscaling_group.rancher_worker.0.name]
+  }
+}
```
Comment (drpebcak) [inline rancher-server/data.tf:67]: Do these wait for the ASGs to reach capacity and for the instances to be ready before it returns?

File: rancher-server/variables.tf [hunk 2]
```diff
@@ -4,10 +4,16 @@ variable "rancher_password" {
...
 variable "rancher_version" {
   type        = string
-  default     = "2.2.8"
+  default     = "2.2.9"
   description = "Version of Rancher to install"
 }
 
+variable "extra_ssh_keys" {
+  type        = list
+  default     = ["ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC06Qvs+Y9JiyOTeYNGAN/Ukq7SmeCTr7EreD1K8Lwu5VuOmo+SBZh685tNTEGV044HgFvGEOBVreDlO2ArYuwHjUBGnpQGV8/abjoeLrmZBdREAUzBQ1h2GFE/WssKUfum81cnigRK1J3tWP7emq/Y2h/Zw5F09yiCIlXMBX2auKWUCXqwG3xKTi1NVSF9N6BGyFolrAR0LZJ6k7UBXPRc/QDTclI427gSJNbnmn8LVym6YxacV/V9Y7s23iR5zYbhLPe9VJWYNk1brVvfUVb3mILVVYz76KGEq8SHdWlPQPCOp+fSJ+PezDRklnex/MmvhNrBOmMSNcpj7wSLA3hD wmaxwell@wmaxwell-laptop", "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIN5O7k6gRYCU7YPkCH6dyXVW10izMAkDAQtQxNxdRE22 drpebcak"]
```
Comment (drpebcak) [inline rancher-server/variables.tf]: Let's remove these. Most people won't want my key ;)

Comment (drpebcak) [reply rancher-server/variables.tf]: Probably should default to `[]`

File: rancher-server/README.md [hunk 3]
```diff
@@ -40,12 +40,13 @@ module "rancher_server" {
 | certmanager\_version | Version of cert-manager to install | string | `"0.10.0"` | no |
 | creds\_output\_path | Where to save the id_rsa config file. Should end in a forward slash `/` . | string | `"./"` | no |
 | domain |  | string | `"eng.rancher.space"` | no |
+| extra\_ssh\_keys | Extra ssh keys to inject into Rancher instances | list | `[ "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC06Qvs+Y9JiyOTeYNGAN/Ukq7SmeCTr7EreD1K8Lwu5VuOmo+SBZh685tNTEGV044HgFvGEOBVreDlO2ArYuwHjUBGnpQGV8/abjoeLrmZBdREAUzBQ1h2GFE/WssKUfum81cnigRK1J3tWP7emq/Y2h/Zw5F09yiCIlXMBX2auKWUCXqwG3xKTi1NVSF9N6BGyFolrAR0LZJ6k7UBXPRc/QDTclI427gSJNbnmn8LVym6YxacV/V9Y7s23iR5zYbhLPe9VJWYNk1brVvfUVb3mILVVYz76KGEq8SHdWlPQPCOp+fSJ+PezDRklnex/MmvhNrBOmMSNcpj7wSLA3hD wmaxwell@wmaxwell-laptop", "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIN5O7k6gRYCU7YPkCH6dyXVW10izMAkDAQtQxNxdRE22 drpebcak" ]` | no |
```
Comment (drpebcak) [inline rancher-server/README.md]: Along with removing my key as a default, let's make sure the doc gets updated 👌

**Review by drpebcak — CHANGES_REQUESTED** [review]

File: rancher-server/data.tf (hunk 1 above)
Comment (jeffb4) [reply rancher-server/data.tf:67]: Yes, by default `aws_autoscaling_group` will wait until the ASG reaches desired capacity before returning, and us using `aws_autoscaling_group.rancher_worker.0.name` here makes an implied dependency in the graph

File: rancher-server/variables.tf (hunk 2 above)
Comment (jeffb4) [reply rancher-server/variables.tf]: Defaulting to []

File: rancher-server/README.md (hunk 3 above)
Comment (jeffb4) [reply rancher-server/README.md]: Docs updated

**Review by drpebcak — APPROVED** [review]

**Pull request merged**